### PR TITLE
DO-358: Create Field type

### PR DESCRIPTION
**Pull request by drewsdav**
## Description
Added a struct type for a form field.

## Motivation and Context
This will help keep a generic structure for a form field when communicating between client and server. I am still unsure if the server or the client should initiate the fetching of a data source, so it is currently on the server to do so.

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have ran any linters on the code being committed.
- [x] My code includes only changes related to the work item.
- [x] My code complies with our [Security checklist](https://github.com/Healthx/DevPatterns/blob/master/Security%20Checklist.md).
- [x] There is no dead, unused code. This includes commented out code.

### CODE REVIEW

File: server/internal/form/field.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package form
+
+// Field defines a structure for fields in a Form object.
+type Field struct {
+	Name        string        `json:"name"`
+	Description string        `json:"description"`
+	Required    bool          `json:"required"`
+	Type        FieldType     `json:"type"`
+	Options     []FieldOption `json:"options"`
+}
```
Comment (achhealthx) [inline server/internal/form/field.go]: Is that the syntax for saying "hey pull this out of Json" when the type comes?  Will you end up nesting structs to parse the json element? (forgive me, I dont know Go .... yet)

File: server/internal/form/field.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+// FieldType defines an "enum" type for what kind of Field is being used.
+type FieldType string
+
+const (
+	TextField      FieldType = "TextField"
+	LargeTextField FieldType = "LargeTextField"
+	Dropdown       FieldType = "Dropdown"
+	Radio          FieldType = "Radio"
+)
```
Comment (achhealthx) [inline server/internal/form/field.go]: Are these grouped together in any way? or how are they referenced?

File: server/internal/form/field.go (hunk 1 above)
Comment (drewsdav) [reply server/internal/form/field.go]: Exactly. Those are field tags. It says, when creating or parsing JSON, use those as the key names. And yes, the structs will be nested. The `FieldType` _should_ be converted to it's string value, and the `Options` array will display as an array in the JSON object.

File: server/internal/form/field.go (hunk 2 above)
Comment (drewsdav) [reply server/internal/form/field.go]: This part is still a bit confusing to me, but these define the values that `FieldType` can take. In the line above, I define a type `FieldType` which is a `string`. These constants just give a name to reference the strings by.

**Pull request merged**